File: 2018-04-04/Services/app/src/main/java/br/ufpe/cin/if710/services/MainThreadService.java
package br.ufpe.cin.if710.services;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;

public class MainThreadService extends Service {


    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        try {
            Thread.sleep(2500);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        return super.onStartCommand(intent, flags, startId);
    }

    @Override
    public IBinder onBind(Intent intent) {
        // TODO: Return the communication channel to the service.
        throw new UnsupportedOperationException("Not yet implemented");
    }
}

File: 2018.1/2018-04-04/Services/app/src/main/java/br/ufpe/cin/if710/services/MainThreadService.kt
package br.ufpe.cin.if710.services

import android.app.Service
import android.content.Intent
import android.os.IBinder

class MainThreadService : Service() {


    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
        try {
            Thread.sleep(2500)
        } catch (e: InterruptedException) {
            e.printStackTrace()
        }

        return super.onStartCommand(intent, flags, startId)
    }

    override fun onBind(intent: Intent): IBinder? {
        // TODO: Return the communication channel to the service.
        throw UnsupportedOperationException("Not yet implemented")
    }
}
